spark-board-ui/src/useDagAnimation/useDagAnimation.js

```diff
@@ -54,25 +59,32 @@ export function useDagAnimation(nodes, edges, targetPositions, setNodes) {
 
         if (hasUpdated) {
             // update the node positions
-            setNodes(nodes.map((node) => {
+            setCurrentPositions(new Map(nodes.map((node) => {
                 if (node.type == "column") {
-                    return node
+                    return [node.id, node.position]
                 }
                 const i = nodeIdMapping.current.get(node.id);
-                return {
-                    ...node,
-                    position: {
-                        x: state.current[0][i * 2],
-                        y: state.current[0][i * 2 + 1]
-                    }
-                }
-            }));
+                return [node.id, {
+                    x: state.current[0][i * 2],
+                    y: state.current[0][i * 2 + 1]
+                }]
+            })));
         }
 
         // wait for the next frame
         animationId.current = requestAnimationFrame(onFrame)
     }
 
+    // update the transformation nodes when the current position changes
+    useEffect(() => {
+        setNodes((nds) => nds.map(node => {
```

Comment:
the real fix was to use a callback inside of `setNodes`, right?

Reply:
The fix is not having a race condition XD